tool.ts

```diff
@@ -207,16 +220,36 @@ export async function downloadTool(url: string, fileName?: string): Promise<stri
             }
 
             // TODO: retries
+            tl.debug('downloading');
+            let response: httpm.HttpClientResponse = await http.get(url);
+
+            // handle HTTP 302 redirect
+            if (response.message.statusCode === httpm.HttpCodes.ResourceMoved
+                && allowRedirect) {
+                tl.debug('redirect received, following Location header');
+                const location: any = response.message.headers["location"];
+                tl.debug('redirect location: ' + location);
+
+                if (location) {
+                    //return await downloadTool(location, fileName);
+                    resolve(downloadTool(location, fileName));
```

Comment:
Please let me know if this is correct. I would have expected the code to be the line before that is commented but that code never resolved the promise and the code hung and failed on timeout. I guess because this is inside the promise creation we have to resolve. Seems like a strange way to write the code.